meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -2,9 +2,9 @@
 
 setenv bootcmd_resetvars 'setenv kernel_image; setenv bootargs; setenv kernel_image2; setenv bootargs2'
 setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize}'
-setenv bootcmd_load_f 'ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} "/boot"${kernel_image}'
-setenv bootcmd_tee_ovy 'if test -n ${optee_ovl_addr}; then imxtract ${loadaddr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file} fdt@@FIT_NODE_SEPARATOR@@${fdt_file} ${fdt_addr}; fdt addr ${fdt_addr}; fdt resize 0x1000; fdt apply ${optee_ovl_addr}; fi'
-setenv bootcmd_run 'bootm ${loadaddr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file} ${loadaddr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file} ${fdt_addr}'
+setenv bootcmd_load_f 'ext4load ${devtype} ${devnum}:${rootpart} ${fit_addr} "/boot"${kernel_image}'
+setenv bootcmd_tee_ovy 'if test -n ${optee_ovl_addr}; then imxtract ${fit_addr}#${fdt_file_final} fdt@@FIT_NODE_SEPARATOR@@${fdt_file_final} ${fdt_addr}; fdt addr ${fdt_addr}; fdt resize 0x1000; fdt apply ${optee_ovl_addr}; fi'
```

Comment:
There is a subtle change at bootcmd_tee_ovy  here:

```
-imxtract ${loadaddr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file} fdt@@FIT_NODE_SEPARATOR@@${fdt_file} ${fdt_addr};
+imxtract ${fit_addr}#${fdt_file_final} fdt@@FIT_NODE_SEPARATOR@@${fdt_file_final} ${fdt_addr}
```

The original change was using the conf node, and that was on purpose because conf is the only thing that gets signed. While your change will work the same way, I believe it won't verify the signature of the fdt_file_final node when not used together with the respective conf, since it is not signed.

An example for https://ci.foundries.io/projects/lmp/builds/1284/apalis-imx6/artifacts/other/fitImage-initramfs-ostree-lmp-image-apalis-imx6-apalis-imx6 (dtc -I dtb -O dts):

The fdt node:
```
               fdt-imx6q-apalis-eval.dtb {
                        description = "Flattened Device Tree blob";
                        type = "flat_dt";
                        arch = "arm";
                        compression = "none";
                        load = <0x15800000>;
                        hash-1 {
                                value = <0x13823165 0x911d9ac1 0x4e231641 0xe73e923d 0x89bdae0e 0xffc1eb9b 0x5925dd9a 0x366b9def>;
                                algo = "sha256";
                        };
                };
```

Conf node:
```
                conf-imx6q-apalis-eval.dtb {
                        description = "1 Linux kernel, FDT blob, ramdisk";
                        kernel = "kernel-1";
                        fdt = "fdt-imx6q-apalis-eval.dtb";
                        ramdisk = "ramdisk-1";

                        hash-1 {
                                algo = "sha256";
                        };

                        signature-1 {
                                hashed-strings = <0x00 0x8e>;
                                hashed-nodes = "/\0/configurations/conf-imx6q-apalis-eval.dtb\0/images/kernel-1\0/images/kernel-1/hash-1\0/images/fdt-imx6q-apalis-eval.dtb\0/images/fdt-imx6q-apalis-eval.dtb/hash-1\0/images/ramdisk-1\0/images/ramdisk-1/hash-1";
                                timestamp = <0x00>;
                                signer-version = "2020.07";
                                signer-name = "mkimage";
                                value = <0x25fe5b29 0x58cd62cd 0xd527e1b8 0x72fd3db7 0xb13280ff 0x4c654c1e 0x3c5eb05a 0x895c3641 0xf2919edc 0x4eed9dda 0xaa4bd1e 0x57d66669 0xabf8112 0xa6db8778 0x3cc76f8 0x535a5683 0xa9615121 0x66613ae3 0x8b5660a2 0x55a12bc1 0x3cc26b98 0xc61d6586 0x7d81486 0xcfbd3881 0xb6281722 0x3568af7 0xe418804b 0x46c94ec9 0x1725ba7f 0xe7c04587 0x9ab505c3 0xbf7afc0b 0x52af6447 0x7437b891 0xc741b244 0xbfebf45f 0x16b37044 0xfc51580c 0x94e250b8 0xbb0adf1d 0xd4eafe61 0x9e02bd44 0x540dcaaf 0x12d9dcc 0xfb0585d1 0x5ca7592 0x15d8a390 0x724b2615 0xc5a956fc 0xec3bdff0 0x314641da 0x207f7604 0x46f96b38 0x4a49ac4c 0xe277d98c 0xc40c058a 0xe9984e2a 0x5893648a 0x8af33c8 0x454b133 0x68a86989 0x659847f0 0xa5df29b7 0x66995ddb>;
                                algo = "sha256,rsa2048";
                                key-name-hint = "dev";
                                sign-images = "kernel\0fdt\0ramdisk";
                        };
                };
```

Reply:
good catch!
I've done that unintentioanlly, fixing now